core/module/services.types.ts

```diff
@@ -54,6 +61,19 @@ export interface AuthService {
 }
 
 export interface SubscriptionService {
-  getSubscriptionLink(): Promise<{ url: string | null }>
-  getActiveSubscription(): Promise<Record<string, string|number>>
-}
\ No newline at end of file
+  getSubscriptionLink(): Promise<{ url: string | null }>;
+  getActiveSubscription(): Promise<Record<string, string | number>>;
+}
+
+export interface TemplateService {
+  getTemplates(page: number, pageSize: number): Promise<PaginatedResponse<Template> | null>
+  getTemplate(id: number): Promise<Template | null>;
```

Comment:
Create an object GenesisTemplate in `core.types`

I don't need or want to know Supabase-specific implementation, but a generic, common an minimum interface for our sitem.